fleetspeak/src/server/https/https_test.go

```diff
@@ -77,7 +77,7 @@ func makeServer(t *testing.T, caseName string) (*server.Server, *sqlite.Datastor
 	return ts.S, ts.DS, tl.Addr().String()
 }
 
-func makeClient(t *testing.T) (common.ClientID, *http.Client) {
+func makeClient(t *testing.T) (common.ClientID, *http.Client, []byte) {
```

Comment:
I'd like the header based tests to behave closer to the use-case if possible, i.e. have it not use TLS and therefore actually missing the certificate. If we add an argument `tls bool` here and have it not include the TLSClientConfig further below be enough to simulate that?

Reply:
Hi @torsm,

I agree that the TLS certificate should be missing when we make the client and perform the transport tests. However, in our use case we still have a TLS connection between the Load Balancer and the Fleetspeak Backend. 

GCP allows Load Balancers to encrypt communication to the backend [Encryption from the load balancer to the backends](https://cloud.google.com/load-balancing/docs/ssl-certificates/encryption-to-the-backends). The problem is that TLS termination between the client to the load balancer drops the client certificate on the floor.